connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -302,12 +302,12 @@ private[kafka010] class KafkaMicroBatchStream(
   }
 
   /**
-   * If `failOnDataLoss` is true, this method will throw an `IllegalStateException`.
+   * If `failOnDataLoss` is true, this method will throw the exception.
    * Otherwise, just log a warning.
    */
-  private def reportDataLoss(message: String): Unit = {
+  private def reportDataLoss(message: String, getException: () => Throwable): Unit = {
```

Comment:
This is scala's `call by name`. I was going to do this, but saw that our style guide asks us to avoid using it.

> Avoid using call by name. Use () => T explicitly.

https://github.com/databricks/scala-style-guide?tab=readme-ov-file#call-by-name

Reply:
oh I see, I wasn't aware of that : ) Thanks for the pointer